###author, forum, title, text, 여태까지 한 것들을 불러와서 하나로 합침

In [1]:
import pandas as pd
import numpy as np
import json
import pickle

max_features = 1000
author_forum_p = pickle.load(open("author, forum_%d.p" % max_features, "rb"))
title_p = pickle.load(open("title_%d.p" % max_features, "rb"))
text_p = pickle.load(open("text_%d.p" % max_features, "rb"))

train = pd.concat([author_forum_p, title_p, text_p], axis = 1)

train

forumid istroll         0         1         2         3         4  \
0            16   False  0.000500  0.000500  0.000500  0.000500  0.000500   
1            31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
2            31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
3            32   False  0.001000  0.001000  0.001000  0.001000  0.001000   
4            23   False  0.001000  0.001000  0.001000  0.001000  0.001000   
5            32   False  0.001000  0.001000  0.001000  0.001000  0.001000   
6            31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
7            23   False  0.001000  0.001000  0.001000  0.001000  0.001000   
8            31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
9            31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
10           32   False  0.001000  0.001000  0.001000  0.001000  0.001000   
11            4   False  0.001000  0.001000  0.001000  0.001000  0.001000   
12           13   False  0.000333  0.000333  0.000333  0.000333  0.000333   
13           31   False  0.000250  0.000250  0.000250  0.000250  0.000250   
14            4   False  0.000333  0.000333  0.000333  0.000333  0.000333   
15           31   False  0.000333  0.000333  0.000333  0.000333  0.000333   
16           13   False  0.000333  0.000333  0.000333  0.000333  0.000333   
17           13   False  0.000333  0.000333  0.000333  0.000333  0.000333   
18           31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
19            4   False  0.000333  0.000333  0.000333  0.000333  0.000333   
20           31   False  0.000333  0.000333  0.000333  0.000333  0.000333   
21           31   False  0.000200  0.000200  0.000200  0.000200  0.000200   
22           32   False  0.000250  0.000250  0.000250  0.000250  0.000250   
23            4   False  0.000333  0.000333  0.000333  0.000333  0.000333   
24           12   False  0.001000  0.001000  0.001000  0.001000  0.001000   
25           32   False  0.001000  0.001000  0.001000  0.001000  0.001000   
26           32   False  0.001000  0.001000  0.001000  0.001000  0.001000   
27           32   False  0.001000  0.001000  0.001000  0.001000  0.001000   
28           31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
29           12   False  0.001000  0.001000  0.001000  0.001000  0.001000   
...         ...     ...       ...       ...       ...       ...       ...   
172014       31   False  0.000500  0.000500  0.000500  0.000500  0.000500   
172015       34    True  0.000500  0.000500  0.000500  0.000500  0.000500   
172016       14   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172017       12   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172018       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172019       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172020       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172021        4   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172022       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172023       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172024       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172025       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172026        5   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172027        1   False  0.000500  0.000500  0.000500  0.000500  0.000500   
172028       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172029        2   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172030       31   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172031       31   False  0.000167  0.000167  0.000167  0.000167  0.000167   
172032       32   False  0.001000  0.001000  0.001000  0.001000  0.001000   
172033       27   False  0.001000  0.001000  0.001000  0.001000  0.001000   
1720

### predictor, model 생성

In [5]:
from sklearn.ensemble import RandomForestClassifier

label = 'istroll'
pre = train.columns.drop([label])

model = RandomForestClassifier(n_estimators=10, n_jobs=6)

###cross validation

In [6]:
from sklearn import cross_validation

cv_value = 3

scores = cross_validation.cross_val_score(model, train[pre], train[label], cv=cv_value, scoring="roc_auc")
cv_result = scores.mean()

print(cv_result)

0.662467774802


##Result

   * ###morphs
   * forumid와 author, title, text를 bag of words로 변환 (feature : 1,000개) - 0.721
   * forumid와 author, title을 bag of words로 변환 (feature : 1,000개) - 0.712577335639
   * forumid와 author, text를 bag of words로 변환 (feature : 1,000개) - 0.721
   
   * ###pos
   * forumid와 author, title, text의 pos를 bag of words로 변환 (feature : 1,000개) - 0.728390543803
   * forumid와 author, title, text의 pos를 LDA로 변환 (keep_n : 5,000, num_topics: 1,000) - 0.662467774802
   * forumid와 author, title의 pos를 bag of words로 변환 (feature : 1,000개) - 0.704123734689
   * forumid와 author, text의 pos를 bag of words로 변환 (feature : 1,000개) - 0.720707529009